doc/configoptions.md

```diff
@@ -5,161 +5,102 @@ in `data/*/*.file_list` (see [file list docu](files.md)).
 
 The options here are environment variables used by `mk_image`. They are typically set in `Makefile`.
 
+Notation: in the reference below, the uppercase FOO is the value of the option
+(environment variable) lowercase foo.
 
 ## General options
 
-- cache=1|2|4
+<!--
+use this to find out actual usage:
+  grep bin/mk_image Makefile | tr ' ' '\n' | sort -u
+-->
 
-  cache the rpms we use, it is actually a bitmask; the most useful value is 4, which
-  will cache the unpacked rpm in a separate directory
+- alternatives=1
 
-- debug=ignore
+  perform a fix-up of update-alternatives links (because...? so that...?)
 
-  ignore noncritical errors
+- disjunct=WORD (eg. `initrd`, `root`)
 
-- debug=account
+  separate WORD from the rest (in what way? so that...?)
 
-  create disk usage statistics, stored in ```tmp/&lt;image_name&gt;.size```
+- filelist=FILELIST
 
-- dist=8.1, dist=8.1-i386+kde
-
-  build for this release
-
-- suserelease=8.0
-
-  obsolete, equivalent to dist=
-
-- kernel=&lt;kernel rpm&gt;
-
-  give kernel package name explicitly
+  use SRC/FILELIST.file_list
 
-- theme=&lt;theme&gt;
+- fs=FS (eg. `dir`, `squashfs`, `cpio`, default: `none`)
 
-  select this theme (default: theme=SuSE)
+  Use FS as the filesystem for the image.
+  May end with `.gz` or `.xz` for compression.
 
-- themes=&lt;list of themes&gt;
+- image=IMAGE (mandatory)
 
-  list of supported themes, support all if empty
+  Use tmp/IMAGE as temporary directory and image/IMAGE as the final image
 
-## root
+- tmpdir=f
 
-- uncompressed_root=1
 
-  build uncompressed image
+- kernel=KERNELRPM (eg. `kernel-default`)
 
-- keeproot=1
+  name of kernel RPM, available as `<kernel_rpm>` in *.file_list
 
-  don't rebuild everything, just recreate (copy the cached tree and compress) the
-  last image (useful to speed up testing)
+- libdeps=DEPS (a comma separated list)
 
-- filelist=foo
+  run `bin/check_libs` on the DEPS (in `tmp/`)
 
-  build using foo.file_list
+- mode=MODE (`add`, `keep`, default: ``)
```

Comment:
mode='add' adds files to the image tmp dir (without deleting it first), mode='keep' keeps all data (does nothing) with the tmp dir data

Reply:
So, the default mode:

1. erases the TMPDIR
2. adds files according to FILELIST
3. packs the TMPDIR with FS into IMAGE

And mode=add skips (1), and mode=keep skips (1) and (2)?

Reply:
yes